ci: pyspark 4.0 test failures (2)

there's 2 test failures from pyspark 4.0
```
FAILED tests/expr_and_series/mean_horizontal_test.py::test_meanh[pyspark-a] - pyspark.errors.exceptions.captured.ArithmeticException: [DIVIDE_BY_ZERO] Division by zero. Use `try_divide` to tolerate divisor being 0 and return NULL instead. If necessary set "spark.sql.ansi.enabled" to "false" to bypass this error. SQLSTATE: 22012
== DataFrame ==
"__truediv__" was called from
/home/runner/work/narwhals/narwhals/narwhals/_spark_like/namespace.py:136
FAILED tests/expr_and_series/mean_horizontal_test.py::test_meanh[pyspark-col_expr0] - pyspark.errors.exceptions.captured.ArithmeticException: [DIVIDE_BY_ZERO] Division by zero. Use `try_divide` to tolerate divisor being 0 and return NULL instead. If necessary set "spark.sql.ansi.enabled" to "false" to bypass this error. SQLSTATE: 22012
== DataFrame ==
"__truediv__" was called from
/home/runner/work/narwhals/narwhals/narwhals/_spark_like/namespace.py:136
```

--- a/narwhals/_spark_like/namespace.py
+++ b/narwhals/_spark_like/namespace.py
@@ -131,18 +131,21 @@ class SparkLikeNamespace(
     def mean_horizontal(self, *exprs: SparkLikeExpr) -> SparkLikeExpr:
         def func(df: SparkLikeLazyFrame) -> list[Column]:
             cols = [c for _expr in exprs for c in _expr(df)]
+            F = exprs[0]._F  # noqa: N806
+            # PySpark before 3.5 doesn't have `try_divide`, SQLFrame doesn't have it.
+            divide = getattr(F, "try_divide", operator.truediv)
             return [
-                (
+                divide(
                     reduce(
                         operator.add, (df._F.coalesce(col, df._F.lit(0)) for col in cols)
-                    )
-                    / reduce(
+                    ),
+                    reduce(
                         operator.add,
                         (
                             col.isNotNull().cast(df._native_dtypes.IntegerType())
                             for col in cols
                         ),
-                    )
+                    ),
                 )
             ]
 

